src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -67,8 +67,8 @@ pub fn collect_trait_impls(krate: Crate, cx: &DocContext<'_>) -> Crate {
         if !def_id.is_local() {
             inline::build_impl(cx, def_id, &mut new_items);
 
-            let auto_impls = get_auto_traits_with_def_id(cx, def_id);
-            let blanket_impls = get_blanket_impls_with_def_id(cx, def_id);
+            let auto_impls = get_auto_traits(cx, def_id, None);
+            let blanket_impls = get_blanket_impls(cx, def_id, None);
```

Comment:
Hmm, this is technically wrong, `def_id` is not even pointing to a type, but rather an `impl`, which `get_def_from_def_id` *happens* to get the `Self` type for, using `type_of` - that, at least, explains the handling of primitive types (I was guessing type aliases, but no).

So I'll need to refactor things a bit more, turns out.